maven-plugin/src/test/java/org/sonatype/nexus/maven/staging/StagingDeployMojoTest.java

```diff
@@ -227,12 +230,17 @@ public void upload() throws Exception {
     assertThat(attributes.get(VERSION_KEY), is(equalTo(VERSION)));
 
     Collection<Asset> assets = component.getAssets();
+    assertThat(assets.size(), is(equalTo(3)));
 
-    assertThat(assets.size(), is(equalTo(2)));
+    // check pom asset
+    List<Asset> pomAssets = assets.stream().filter(this::isPomAsset).collect(toList());
+    assertThat(pomAssets.size(), is(equalTo(1)));
+    assertThat(pomAssets.get(0).getFilename(), is(equalTo(getPom().getName())));
+    assertThat(pomAssets.get(0).getData(), notNullValue());
 
-    for (Asset asset : assets) {
+    // check all other assets
+    for (Asset asset : assets.stream().filter(asset -> !isPomAsset(asset)).collect(toList())) {
```

Comment:
minor: seems like now that you're streaming / filtering, it might be cleaner to have:
```
assets.stream().filter(this::isPomAsset).foreach(asset -> {
```
(I didn't check the above syntax, so caveat emptor)

Reply:
Unless you feel strongly, I am gonna leave this.

I tried some tests, but I end up writing my own `not` method, and more code. In java 11 it is part of the `Predicate`. For reference: https://stackoverflow.com/questions/21488056/how-to-negate-a-method-reference-predicate

Reply:
I don't mind you leaving it.